partiql-parser/src/parse/parser_state.rs

```diff
@@ -0,0 +1,83 @@
+use crate::error::ParseError;
+use crate::parse::lexer;
+use std::ops::Range;
+
+use partiql_ast::ast;
+
+use lalrpop_util::ErrorRecovery;
+
+use partiql_ast::ast::{AstNode, NodeId};
+
+use partiql_source_map::location::{ByteOffset, BytePosition, Location};
+use partiql_source_map::metadata::LocationMap;
+
+type ParseErrorRecovery<'input> =
+    ErrorRecovery<ByteOffset, lexer::Token<'input>, ParseError<'input, BytePosition>>;
+type ParseErrors<'input> = Vec<ParseErrorRecovery<'input>>;
+
+const INIT_LOCATIONS: usize = 50;
+
+pub trait IdGenerator {
+    fn id(&mut self) -> NodeId;
+}
+
+pub(crate) struct NodeIdGenerator {
+    next_id: ast::NodeId,
+}
```

Comment:
Could use some docs for these public apis (`IdGenerator` and `NodeIdGenerator`). Also for `ParserState` on L42

Reply:
I'll add docs.

But these aren't really public apis. Though marked `pub`, they are in a module that is private included in its `super`, so they are public to the `super`, but not the outside world.

You can see what's exported with `cargo doc --open`.